MKS Robin Pro with a MKS Robin TFT35 will not compile with color ui or lvgl ui selected

### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

I set the environment in PlatformIO to mks_robin_pro. I copy the latest configuration.h example for the mks pro from bugfix to the Marlin folder. Open folder in vscode/platformIO.  set the tft to Robin_tft35. set UI to either Color or lvgl and it will not compile. no other changes have been made.

### Bug Timeline

it reacted the same way in 2.07, but it was a new printer, and 2.08 came out days after I got it.

### Expected behavior

I expected it to compile with either of the chosen UI, it will compile if you choose the tft_classical.

### Actual behavior

I got an error compiling saying pins weren't defined for this scope. Compiling .pio\build\mks_robin_pro\src\src\HAL\shared\servo.cpp.o
Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp: In static member function 'static void TFT_FSMC::Init()':
Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp:89:16: error: 'FSMC_CS_PIN' was not declared in this scope
   uint8_t cs = FSMC_CS_PIN, rs = FSMC_RS_PIN;
                ^~~~~~~~~~~
Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp:89:16: note: suggested alternative: 'FSMC_CS_NE4'
   uint8_t cs = FSMC_CS_PIN, rs = FSMC_RS_PIN;
                ^~~~~~~~~~~
                FSMC_CS_NE4
Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp:115:11: error: 'rs' was not declared in this scope
   switch (rs) {
           ^~
Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp:115:11: note: suggested alternative: 'cs'
   switch (rs) {
           ^~
           cs
Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp:172:25: error: 'rs' was not declared in this scope
   gpio_set_mode(PIN_MAP[rs].gpio_device, PIN_MAP[rs].gpio_bit, GPIO_AF_OUTPUT_PP);  //FSMC_RS_Ax
                         ^~
Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp:172:25: note: suggested alternative: 'cs'
   gpio_set_mode(PIN_MAP[rs].gpio_device, PIN_MAP[rs].gpio_bit, GPIO_AF_OUTPUT_PP);  //FSMC_RS_Ax
                         ^~
                         cs

### Steps to Reproduce

1. Download latest bugfix or 2.08. expand
2. Download the latest configuration files. expand
3. Copy the configuration for the mks_robin_pro to Marlin
4. open VSCode and PlatformIO
5. In PlatformIO open folder and select Marlin folder.
6. Change environment for PlatformIO to mks_robin_pro. save file
7. Change configuration.h change from generic_tft to robin_tft35. Choose either color_ui or lvgl_ui and save file.
8. attempt compile

### Version of Marlin Firmware

2.08

### Printer model

JGMaker Artist-D Pro

### Electronics

stock. mks_robin_pro with mks_robin_tft35, tmc2209_standalone

### Add-ons

none

### Your Slicer

Simplify3D

### Host Software

SD Card (headless)

### Additional information & file uploads

[MarlinConfig.zip](https://github.com/MarlinFirmware/Marlin/files/6462530/MarlinConfig.zip)

--- a/Marlin/src/pins/stm32f1/pins_MKS_ROBIN_PRO.h
+++ b/Marlin/src/pins/stm32f1/pins_MKS_ROBIN_PRO.h
@@ -43,7 +43,7 @@
 //
 // Note: MKS Robin board is using SPI2 interface.
 //
-#define SPI_DEVICE                             2
+#define SPI_DEVICE                          2
 
 //
 // Servos
@@ -188,7 +188,7 @@
 // SD Card
 //
 #ifndef SDCARD_CONNECTION
-  #define SDCARD_CONNECTION              ONBOARD
+  #define SDCARD_CONNECTION                 ONBOARD
 #endif
 
 #if SD_CONNECTION_IS(LCD)
@@ -211,15 +211,15 @@
  * to let the bootloader init the screen.
  */
 #if HAS_FSMC_TFT
-  #define FSMC_CS_PIN                       PD7   // NE4
-  #define FSMC_RS_PIN                       PD11  // A0
-  #define TFT_CS_PIN                 FSMC_CS_PIN
-  #define TFT_RS_PIN                 FSMC_RS_PIN
+  #define FSMC_CS_PIN                       PD7           // NE4
+  #define FSMC_RS_PIN                       PD11          // A0
+  #define TFT_CS_PIN                        FSMC_CS_PIN
+  #define TFT_RS_PIN                        FSMC_RS_PIN
 
   #define LCD_RESET_PIN                     PF6
   #define LCD_BACKLIGHT_PIN                 PD13
-  #define TFT_RESET_PIN            LCD_RESET_PIN
-  #define TFT_BACKLIGHT_PIN    LCD_BACKLIGHT_PIN
+  #define TFT_RESET_PIN                     LCD_RESET_PIN
+  #define TFT_BACKLIGHT_PIN                 LCD_BACKLIGHT_PIN
 
   #if NEED_TOUCH_PINS
     #define TOUCH_CS_PIN                    PA7
@@ -255,7 +255,8 @@
     #define DOGLCD_SCK                      PB13
     #define DOGLCD_MOSI                     PB15
 
-  #else                                           // !MKS_MINI_12864 && !ENDER2_STOCKDISPLAY
+  #else
+  // !MKS_MINI_12864 && !ENDER2_STOCKDISPLAY
 
     #define LCD_PINS_D4                     PF14
     #if IS_ULTIPANEL
@@ -264,7 +265,7 @@
       #define LCD_PINS_D7                   PF13
 
       #if ENABLED(REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER)
-        #define BTN_ENC_EN           LCD_PINS_D7  // Detect the presence of the encoder
+        #define BTN_ENC_EN                  LCD_PINS_D7  // Detect the presence of the encoder
       #endif
 
     #endif
@@ -274,19 +275,19 @@
 #endif
 
 #ifndef BOARD_ST7920_DELAY_1
-  #define BOARD_ST7920_DELAY_1     DELAY_NS(125)
+  #define BOARD_ST7920_DELAY_1              DELAY_NS(125)
 #endif
 #ifndef BOARD_ST7920_DELAY_2
-  #define BOARD_ST7920_DELAY_2     DELAY_NS(125)
+  #define BOARD_ST7920_DELAY_2              DELAY_NS(125)
 #endif
 #ifndef BOARD_ST7920_DELAY_3
-  #define BOARD_ST7920_DELAY_3     DELAY_NS(125)
+  #define BOARD_ST7920_DELAY_3              DELAY_NS(125)
 #endif
 
-#define HAS_SPI_FLASH                          1
+#define HAS_SPI_FLASH                       1
 #if HAS_SPI_FLASH
-  #define SPI_FLASH_SIZE               0x1000000  // 16MB
-  #define W25QXX_CS_PIN                     PB12  // Flash chip-select
+  #define SPI_FLASH_SIZE                    0x1000000   // 16MB
+  #define W25QXX_CS_PIN                     PB12        // Flash chip-select
   #define W25QXX_MOSI_PIN                   PB15
   #define W25QXX_MISO_PIN                   PB14
   #define W25QXX_SCK_PIN                    PB13

--- a/Marlin/src/pins/stm32f1/pins_MKS_ROBIN_PRO.h
+++ b/Marlin/src/pins/stm32f1/pins_MKS_ROBIN_PRO.h
@@ -210,12 +210,16 @@
  * If the screen stays white, disable 'LCD_RESET_PIN'
  * to let the bootloader init the screen.
  */
-#if HAS_FSMC_GRAPHICAL_TFT
+#if HAS_FSMC_TFT
   #define FSMC_CS_PIN                       PD7   // NE4
   #define FSMC_RS_PIN                       PD11  // A0
+  #define TFT_CS_PIN                 FSMC_CS_PIN
+  #define TFT_RS_PIN                 FSMC_RS_PIN
 
   #define LCD_RESET_PIN                     PF6
   #define LCD_BACKLIGHT_PIN                 PD13
+  #define TFT_RESET_PIN            LCD_RESET_PIN
+  #define TFT_BACKLIGHT_PIN    LCD_BACKLIGHT_PIN
 
   #if NEED_TOUCH_PINS
     #define TOUCH_CS_PIN                    PA7

--- a/Marlin/src/pins/stm32f1/pins_MKS_ROBIN_PRO.h
+++ b/Marlin/src/pins/stm32f1/pins_MKS_ROBIN_PRO.h
@@ -282,3 +282,12 @@
 #ifndef BOARD_ST7920_DELAY_3
   #define BOARD_ST7920_DELAY_3     DELAY_NS(125)
 #endif
+
+#define HAS_SPI_FLASH                          1
+#if HAS_SPI_FLASH
+  #define SPI_FLASH_SIZE               0x1000000  // 16MB
+  #define W25QXX_CS_PIN                     PB12  // Flash chip-select
+  #define W25QXX_MOSI_PIN                   PB15
+  #define W25QXX_MISO_PIN                   PB14
+  #define W25QXX_SCK_PIN                    PB13
+#endif

